visionmetrics/caption/azure_openai_model_eval_base.py

```diff
@@ -0,0 +1,161 @@
+        self.prompt_template = prompt_template
+        self.system_message = system_message
+        self.positive_threshold = positive_threshold
+        self.negative_value = negative_value
+
+        if num_responses > 1:
+            raise NotImplementedError("This metric currently only supports single-response scoring.")
+        self.num_responses = num_responses
+
+        self.model = OpenAITextChatModel(endpoint=endpoint, deployment_name=deployment_name, api_key=None, temperature=temperature, max_tokens=max_tokens,
+                                         requests_inverval=requests_interval, num_responses=num_responses, delimiter=MULTIPLE_RESPONSES_DELIMITER, system_message=system_message)
+        
+        self.add_state("predictions", default=[], dist_reduce_fx="cat")
+        self.add_state("targets", default=[], dist_reduce_fx="cat")
+        self.add_state("raw_scores", default=[], dist_reduce_fx="cat")
+        self.add_state("scores", default=[], dist_reduce_fx="cat")
+        self.add_state("score_parse_failures", default=torch.tensor(0), dist_reduce_fx="sum")
+        self.add_state("result_status_types", default=[], dist_reduce_fx="cat")
+
+    def _get_numeric_score(self, raw_score: str):
+        """
+        Parses the numeric score value from the string `raw_score`; if not parseable, returns 0.
+        """
+        score = 0.
+        try:
+            score = float(raw_score)
+        except ValueError:
+            self.score_parse_failures += 1
+        return score
+
+    def update(self, predictions, targets):
+        """ Evaluate list of predicted results using OpenAI text model.
+        Args:
+            predictions: list of string predictions [text1, text2, ...], shape: (N, ), type: string
+            targets: list of one or more string ground truth values: [[gt1, gt2, ...], [gt1, gt2, ...], ...], type: string
+                Note: with multiple targets, a good match with any of them is considered correct.
+            multiple_target_join_method: string specifying the joining method to join multiple reference ground truth values in the prompt, either 'any' or 'all'
+        """
+        if len(predictions) != len(targets):
+            raise ValueError("Please update with an equal number of predictions and targets.")
+        
+        self.predictions += predictions
+        self.targets += targets
+        
+        for prediction, target in zip(predictions, targets):
+            prompt = self.prompt_template.replace(PREDICTION_PLACEHOLDER, prediction)
+            final_target = OR_SEPARATOR.join(target)
+            prompt = prompt.replace(TARGET_PLACEHOLDER, final_target)
+            
+            try:
+                result = ["1.0"] if prediction == target else self.model.forward(([prompt], [[]]))
+                result = result[0] if result else ''
+            except:
+                result = '0.0'
```

Comment:
Any particular reason why result here = '0.0' and not '' (empty str)
Also, line 99 may never fail since the chat completion model in irisml already catches exception and returns empty str.

Reply:
Good catch, thanks! Simplified this logic.